src/test/java/seedu/address/logic/commands/AddCommandTest.java

```diff
@@ -0,0 +1,195 @@
+        ModelStubAcceptingApplicationAdded modelStub = new ModelStubAcceptingApplicationAdded();
+        Application validApplication = new ApplicationBuilder().build();
+
+        CommandResult commandResult = new AddCommand(validApplication).execute(modelStub);
+
+        assertEquals(String.format(AddCommand.MESSAGE_SUCCESS, validApplication), commandResult.getFeedbackToUser());
+        assertEquals(Arrays.asList(validApplication), modelStub.applicationsAdded);
+    }
+
+    @Test
+    public void execute_duplicateApplication_throwsCommandException() {
+        Application validApplication = new ApplicationBuilder().build();
+        AddCommand addCommand = new AddCommand(validApplication);
+        ModelStub modelStub = new ModelStubWithApplication(validApplication);
+
+        assertThrows(CommandException.class,
+                AddCommand.MESSAGE_DUPLICATE_APPLICATION, () -> addCommand.execute(modelStub));
+    }
+
+    @Test
+    public void equals() {
+        Application google = new ApplicationBuilder().withCompany("Google").build();
+        Application shopee = new ApplicationBuilder().withCompany("Shopee").build();
+        AddCommand addGoogleCommand = new AddCommand(google);
+        AddCommand addShopeeCommand = new AddCommand(shopee);
+
+        // same object -> returns true
+        assertTrue(addGoogleCommand.equals(addGoogleCommand));
+
+        // same values -> returns true
+        AddCommand addGoogleCommandCopy = new AddCommand(google);
+        assertTrue(addGoogleCommand.equals(addGoogleCommandCopy));
+
+        // different types -> returns false
+        assertFalse(addGoogleCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(addGoogleCommand.equals(null));
+
+        // different application -> returns false
+        assertFalse(addGoogleCommand.equals(addShopeeCommand));
+    }
+
+    /**
+     * A default model stub that has all of the methods failing.
+     */
+    private class ModelStub implements Model {
+        @Override
+        public void setUserPrefs(ReadOnlyUserPrefs userPrefs) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public ReadOnlyUserPrefs getUserPrefs() {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public GuiSettings getGuiSettings() {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void setGuiSettings(GuiSettings guiSettings) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public Path getApplicationBookFilePath() {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void setApplicationBookFilePath(Path addressBookFilePath) {
```

Comment:
One minor error here
```suggestion
        public void setApplicationBookFilePath(Path applicationBookFilePath) {
```